magicblock-chainlink/src/chainlink/fetch_cloner.rs

```diff
@@ -864,7 +887,7 @@ where
         let mut missing_delegation_record = vec![];
 
         // We remove all new subs for accounts that were not found or already in the bank
-        let (accounts_to_clone, record_subs) = {
+        let (mut accounts_to_clone, record_subs) = {
```

Comment:
I'm not sure why this has to be `mut` and the ATA logic runs separately below instead of as part of this block.
Is it plainly for code organization?

Reply:
Mainly yes. I wanted to keep that part of the logic self-contained. It could be refactored, but honestly I think the entire fetch_cloner.rs should be refactored.